src/AutoMerge/util.jl

```diff
@@ -26,6 +22,48 @@ function _clone_repo_into_dir(url::AbstractString, repo_dir)
     return repo_dir
 end
 
+"""
+    load_files_from_url_and_tree_hash(destination::String, url::String, tree_hash::String) -> Bool
+
+Attempts to clone a git repo from `url` into a temporary directory and extract the files and
+folders from a given `tree_hash`, placing them in `destination`.
+
+Returns a boolean indicating if the cloning succeeded.
+"""
+function load_files_from_url_and_tree_hash(destination::String, url::String, tree_hash::String)
+    pkg_clone_dir = mktempdir()
+    clone_success = try
+        _clone_repo_into_dir(url, pkg_clone_dir)
+        true
+    catch e
+        @error "Cloning $url failed"
+        false
+    end
+    Tar.extract(Cmd(`git archive $tree_hash`; dir=pkg_clone_dir), destination)
```

Comment:
The fact that `Tar.extract` takes a `Cmd` as input is really nice!

Reply:
@StefanKarpinski (for both Tar.jl and `Cmd` 😂)